src/hotspot/os/posix/perfMemory_posix.cpp

```diff
@@ -1205,9 +1205,10 @@ static void mmap_attach_shared(int vmid, char** addr, size_t* sizep, TRAPS) {
   FREE_C_HEAP_ARRAY(char, dirname);
   FREE_C_HEAP_ARRAY(char, filename);
 
-  if (fd == OS_ERR || HAS_PENDING_EXCEPTION) {
+  if (fd == OS_ERR) {
     return;
   }
+  assert(!HAS_PENDING_EXCEPTION, "must be");
```

Comment:
I think this is better, so you don't need to read the rather convoluted code in open_sharedmem_file() to understand what the API contract is.
```
if (HAS_PENDING_EXCEPTION) {
  assert(fd == OS_ERR, "open_sharedmem_file always returns OS_ERR on exceptions");
}
if (fd == OS_ERR) {
  return;
}
```